api/src/com/cloud/network/Networks.java

```diff
@@ -251,6 +252,10 @@ public static URI fromString(String candidate) {
                 if (com.cloud.dc.Vlan.UNTAGGED.equalsIgnoreCase(candidate)) {
                     return Native.toUri(candidate);
                 }
+                if (UuidUtils.validateUUID(candidate)){
+                    //Supporting lswitch uuid as vlan id: set broadcast_uri = null and add row on nicira_nvp_router_map with logicalrouter_uuid = candidate
+                    return null;
```

Comment:
One other thing, isn't there a method that validates a UUID? 

Maybe I'm missing something, but wound't `UuidUtils.validateUUID(candidate)` tell you if candidate is a UUID or not?
If so, why not use that instead of the construction with `try { Long.parse(...); } catch (NumberFormatException nfe) { (...) }`?